### PR TITLE
Use bucket backend for static/media

### PR DESCRIPTION
**Pull request by devincowan**
### Pull Request Checklist: 
- [ ] Positive Test Case Written by Dev

- [ ] Automated Testing

- [ ] Sufficient User and Developer Documentation

- [ ] Passing Jenkins Build

- [ ] Peer Code review and approval

### Positive Test Case
1. [Enter positive test case here]

### CODE REVIEW

**Comment by devincowan** [comment]
jenkins please test again

File: hydroshare/urls.py [hunk 1]
```diff
@@ -201,12 +201,14 @@
 ]
 
 # robots.txt URLs for django-robots
+# TODO test robots.txt with bucket
 urlpatterns += [
     url(r"^robots\.txt", include("robots.urls")),
 ]
```
Comment (devincowan) [inline hydroshare/urls.py:207]: I'm not sure how robots.txt will work with static bucket

**Comment by devincowan** [comment]
@sblack-usu this is pretty much ready for review
there is a remaining question about if/how robots.txt will work

File: hydroshare/urls.py (hunk 1 above)
Comment (devincowan) [reply hydroshare/urls.py:207]: I did some checks, believe this will be fine.

File: hs_core/migrations/0080_baseresource_quota_holder.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+    dependencies = [
+        migrations.swappable_dependency(settings.AUTH_USER_MODEL),
+        ('hs_core', '0079_baseresource_bag_last_downloaded'),
+    ]
+
+    operations = [
+        migrations.AddField(
+            model_name='baseresource',
+            name='quota_holder',
```
Comment (sblack-usu) [inline hs_core/migrations/0080_baseresource_quota_holder.py:18]: Is some of the quota mixed in with this PR?

File: hs_discover/views.py [hunk 3]
```diff
@@ -238,7 +238,7 @@ def get(self, request, *args, **kwargs):
                 "title": result.title,
                 "link": result.absolute_url,
                 "availability": result.availability,
-                "availabilityurl": "/static/static/img/{}.png".format(result.availability[0]),
+                "availabilityurl": f"{settings.STATIC_URL}img/{result.availability[0]}.png",
```
Comment (sblack-usu) [inline hs_discover/views.py:241]: nice catch

File: hs_discover/vue.config.js [hunk 4]
```diff
@@ -2,6 +2,13 @@ module.exports = {
...
-  publicPath: '/static/static',
+  outputDir: "templates/hs_discover/",
+  // Here profiding the ability to override the publicPath via an environment variable
+  // Example export VUE_APP_BUCKET_URL_PUBLIC_PATH=http://my-public-bucket && npm run build
+  // This should be the same as the STATIC_URL in the Django settings
+  // https://cli.vuejs.org/config/#publicpath
+  publicPath: process.env.VUE_APP_BUCKET_URL_PUBLIC_PATH ? process.env.VUE_APP_BUCKET_URL_PUBLIC_PATH : "/static/static",
+  // https://cli.vuejs.org/config/#crossorigin
+  // TODO: figure out how to get crossorigin to work with the Django static files
+  crossorigin: "anonymous",
```
Comment (sblack-usu) [inline hs_discover/vue.config.js:13]: noting

File: hydroshare/settings.py [hunk 5]
```diff
@@ -222,7 +222,9 @@ def _(s): return s # noqa
 STATICFILES_FINDERS = (
     "django.contrib.staticfiles.finders.FileSystemFinder",
     "django.contrib.staticfiles.finders.AppDirectoriesFinder",
-    "django.contrib.staticfiles.finders.DefaultStorageFinder",
+    # "django.contrib.staticfiles.finders.DefaultStorageFinder",
+    # We disable the DefaultStorageFinder because otherwise it will search for static files in Google Cloud Storage
+    # https://docs.djangoproject.com/en/3.2/ref/settings/#staticfiles-finders
```
Comment (sblack-usu) [inline hydroshare/settings.py:227]: 🙏

File: hs_discover/vue.config.js (hunk 4 above)
Comment (devincowan) [reply hs_discover/vue.config.js:13]: removed thanks

File: hydroshare/storage.py [hunk 6]
```diff
@@ -1,8 +1,9 @@
-from django.contrib.staticfiles.storage import ManifestStaticFilesStorage
+from storages.backends.gcloud import GoogleCloudStorage
+from django.contrib.staticfiles.storage import ManifestFilesMixin, ManifestStaticFilesStorage
 import logging
 
 
-class ForgivingManifestStaticFilesStorage(ManifestStaticFilesStorage):
+class ForgivingManifestFilesMixin(ManifestFilesMixin):
```
Comment (sblack-usu) [inline hydroshare/storage.py:6]: I love that forgiveness is in the heart of the storage system!

File: theme/templates/accounts/profile.html [hunk 7]
```diff
@@ -662,13 +662,13 @@ <h3 class="panel-title"><span
                                             <tr>
                                                 <td>
                                                     {% if name == "GoogleScholarID" %}
-                                                    <img class="identifier-icon ai ai-google-scholar-square"/>
+                                                    <img class="identifier-icon ai ai-google-scholar-square" src="{% static 'img/google-scholar.svg' %}"/>
```
Comment (sblack-usu) [inline theme/templates/accounts/profile.html:665]: Why are these changes made? How did this work before you introduced source?

**Review by sblack-usu — APPROVED** [review]

File: theme/templates/accounts/profile.html (hunk 7 above)
Comment (devincowan) [reply theme/templates/accounts/profile.html:665]: solid question, it's not obvious!

Before, we had the images defined in our css, with hard links to `static/static`:
https://github.com/hydroshare/hydroshare/pull/5482/files#diff-94565d8a46949822da99c6938042a6414adf62d4d327dc337836b3d20f3f5727L69-L86

Getting an argument for the static url into css requires a whole bunch more in our pipeline (css bundler, sass, etc) so instead, I define the `src` for the image in the django template.

**Pull request merged**